tests/unit/llms/test_llm_loader.py

```diff
@@ -0,0 +1,20 @@
+"""Unit tests for LLMLoader class."""
+
+import pytest
+
+from ols.src import constants
+from ols.src.llms.llm_loader import LLMLoader
+
+
+def test_constructor_no_provider():
+    """Test that constructor checks for provider."""
+    # we use bare Exception in the code, so need to check
+    # message, at least
+    with pytest.raises(Exception, match="ERROR: Missing provider"):
+        LLMLoader(provider=None)
+
+
+def test_constructor_wrong_provider():
+    """Test how wrong provider is checked."""
+    # currently, it just logs error and that's it
+    LLMLoader(provider="foo-bar-baz", model=constants.GRANITE_13B_CHAT_V1)
```

Comment:
```suggestion
    LLMLoader(provider="invalid-provider", model=constants.GRANITE_13B_CHAT_V1)
```

makes it clearer what the expected result is.

Reply:
Also for something like this:

1) separate PR from fixing the message template
2) would it be that hard to add error handling/exception raising in the llm_loader for this so we can have a meaningful test?

Reply:
fixed the provider name @bparees 

Ok so is the expected behavior to raise an exception immediately the unknown provider is ... provided?

Reply:
okay make sense, I'm on it

Reply:
better now @bparees ? :)